### ISSUE TITLE
improvements to calibration functions

### ISSUE THREAD

**Issue by jfree-man** [canmod/macpan2#176]
`mp_simulator`/`mp_tmb_calibrator`

- [x] you can provide `outputs` that don't exist in the model spec, and no warning/error is produced

```
sir = mp_tmb_library("starter_models", "sir", package = "macpan2")

# should be lower case I, "infection"
sir_sim = mp_simulator(sir, time_steps = 5, outputs = c("Infection")) |> mp_trajectory()

# you can similarly do so in mp_tmb_calibrator()
sir_sim = mp_simulator(sir, time_steps = 5, outputs = c("I")) |> mp_trajectory()
sir_calib = mp_tmb_calibrator(sir,
                              , data = sir_sim
                              , traj = "I"
                              , par = "beta"
                              , outputs = c("E","I") # E doesn't exist
                              )

sir_traj = (sir_calib
      |> mp_trajectory()
)
(sir_traj
  |> ggplot(aes(time, value))
  + facet_wrap(vars(matrix), scales='free')
  + geom_line()
)

```

`mp_tmb_calibrator`

- [x]  more informative error message when providing `traj` variable names that don't exist in the model or don't match observed data trajectory name

```
sir_sim = mp_simulator(sir, time_steps = 5, outputs = c("infection")) |> mp_trajectory()

# doesn't match sir_sim trajectory name
sir_calib = mp_tmb_calibrator(sir,
  , data = sir_sim
  , traj = "recovery"
  , par = "beta")

# traj doesn't exist in model spec
sir_calib = mp_tmb_calibrator(sir,
  , data = sir_sim
  , traj = "Infection"
  , par = "beta")

```
- [x] more informative error message when providing `par` variable names that don't exist in the model 

```
sir_calib = mp_tmb_calibrator(sir,
  , data=sir_sim
  , traj = "infection"
  , par = "Beta"
)
# tells me Beta isn't a scalar
```

I'll add additional comments below, if I find anything else.

**Comment by stevencarlislewalker** [comment]
This is fantastic.  Please keep these cases coming.  Have you tried doing stuff with uneven time-steps?

**Comment by jfree-man** [comment]
Yes, it's in the shiver model calibration_example.R , but I am still working on it. I need to understand if it's doing what I think it is.

**Comment by stevencarlislewalker** [comment]
Let me know if you want to chat.

**Comment by stevencarlislewalker** [comment]
First issue: https://github.com/canmod/macpan2/commit/7b32c62e034a558b0bbbf6d3a0a9cdc20f0cd9a7

**Comment by jfree-man** [comment]
`mp_tmb_calibrator`/`mp_trajectory`

- [ ] seg fault, when `time` in observed data is not an integer vector
```
sir_obs = (read.csv(
  "inst/starter_models/shiver/data/hospitalizations_ontario.csv"
  , row.names = NULL
  ) 
  |> rename(time = date)
  |> group_by(time)
  |>  summarize(value = sum(icu_unvac, icu_partial_vac, icu_full_vac,
                          hospitalnonicu_unvac, hospitalnonicu_partial_vac, 
                          hospitalnonicu_full_vac))
  |>  ungroup()
  |>  mutate(matrix="I") # not actually I, just using as an example
)

sir = mp_tmb_library("starter_models", "sir", package = "macpan2")

# get a lapply warning 
sir_calib  = mp_tmb_calibrator(sir,
                               , data=sir_obs
                               , traj = "I"
                               , par = "beta")
# dates are there
sir_calib$cal_spec$integers$obs_times_I

# this seg faults
# mp_trajectory(sir_calib)


```

**Comment by stevencarlislewalker** [comment]
Times as times segfault fixed: 15073c8e223f4ba05e852490f8463a09ca81f556

**Comment by stevencarlislewalker** [comment]
Bad trajectory message fixed: f8a76742f95fe835829a624c1f4927f055804b42

**Comment by stevencarlislewalker** [comment]
Bad par message fixed: df373bb8d70f03d8e4194769b2de7b8611ee1575

**Comment by stevencarlislewalker** [comment]
Fixed everything here.
